src/promptflow/promptflow/azure/_restclient/flow_service_caller.py

```diff
@@ -500,11 +500,23 @@ def create_flow_session(
         sleep_period = 5
         status = None
         timeout_seconds = SESSION_CREATION_TIMEOUT_SECONDS
+        # polling timeout, if user set SESSION_CREATION_TIMEOUT_SECONDS in environment var, use it
+        if os.environ.get(SESSION_CREATION_TIMEOUT_ENV_VAR):
+            try:
+                timeout_seconds = float(os.environ.get(SESSION_CREATION_TIMEOUT_ENV_VAR))
+            except ValueError:
+                raise UserErrorException(
+                    "Environment variable {} with value {} set but failed to parse. "
+                    "Please reset the value to a number.".format(
+                        SESSION_CREATION_TIMEOUT_ENV_VAR, os.environ.get(SESSION_CREATION_TIMEOUT_ENV_VAR)
+                    )
+                )
         # InProgress is only known non-terminal status for now.
         while status in [None, "InProgress"]:
             if time_run + sleep_period > timeout_seconds:
-                message = f"Timeout for session {action} {session_id} for {AUTOMATIC_RUNTIME}.\n" \
-                          "Please resubmit the flow later."
+                message = f"Polling timeout for session {session_id} {action} for {AUTOMATIC_RUNTIME}.\n" \
+                          f"To proceed the {action} for {AUTOMATIC_RUNTIME}, you can retry using the same flow, " \
```

Comment:
Shall we inform how long the timeout peried was? Like the value of above `timeout_seconds`

Reply:
by the way, should we sync this value with real session timeout?